guava/src/com/google/common/escape/CharEscaper.java

```diff
@@ -93,14 +98,21 @@ public String escape(String string) {
    * @return the escaped form of {@code string}
    * @throws NullPointerException if {@code string} is null
    */
-  protected final String escapeSlow(String s, int index) {
-    int slen = s.length();
+  @SuppressWarnings(value = {"argument.type.incompatible",//(141, 49): r.length is required to be @LTLength(value = "r", "dest", offset = "destIndex - 1)
+          // which can be written as @LTLengthOf(value = "dest", offset = "- escaped.length + 1") int destIndexCopy = destIndex;
+          // Moreover, rlen = r.length should be inferred as @LTLengthOf(value = "dest", offset = "-1") already
+          "assignment.type.incompatible"//(133, 59): int `destIndex` is required to be @LessThan("destSizeGrow + 1") to match `destIndexCopy`
+          //but destSizeGrow is declared after when destSize < sizeNeeded
+  })
+  protected final String escapeSlow(String s, @IndexOrHigh("#1") int index) {
+    @LengthOf("s") int slen = s.length();
 
     // Get a destination buffer and setup some loop variables.
     char[] dest = Platform.charBufferFromThreadLocal();
-    int destSize = dest.length;
-    int destIndex = 0;
-    int lastEscape = 0;
+    @LengthOf("dest") int destSize = dest.length;
+    @LTEqLengthOf("dest") int destIndex = 0;
+    int destIndexInternal = destIndex;
```

Comment:
@NhatDinh, do not change the behavior of the code being annotated. That is very important.
Similarly to `UnicodeEscaper.escapeSlow`, some occurences of `destIndex` are replaced by `destIndexInternal`, some are not. That is most likely to work incorrectly.